providers/google/src/airflow/providers/google/cloud/operators/gen_ai.py

```diff
@@ -387,3 +396,809 @@ def execute(self, context: Context):
+        self.hook = GenAIGeminiAPIHook(
+            gcp_conn_id=self.gcp_conn_id,
+            impersonation_chain=self.impersonation_chain,
+            gemini_api_key=self.gemini_api_key,
+        )
+
+        jobs_list = self.hook.list_batch_jobs(list_batch_jobs_config=self.list_batch_jobs_config)
+
+        job_names = []
+        job_objs = []
+
+        try:
+            for job in jobs_list:
+                job_names.append(job.name)
+                job_objs.append(dict(job))
+        except RuntimeError:
+            self.log.info("%s jobs found", len(job_names))
+
+        context["ti"].xcom_push(key="job_names", value=job_names)
+
+        return job_objs
+
+
+class GenAIGeminiDeleteBatchJobOperator(GoogleCloudBaseOperator):
+    """
+    Delete Batch job using Gemini API.
+
+    :param project_id: Required. The ID of the Google Cloud project that the service belongs to.
+    :param location: Required. The ID of the Google Cloud location that the service belongs to.
+    :param model: Required. The name of the publisher model to use for Batch job.
+    :param gemini_api_key: Required. Key to interact with Gemini Batch API.
+    :param job_name: Required. Name of the batch job.
+    :param gcp_conn_id: The connection ID to use connecting to Google Cloud.
+    :param impersonation_chain: Optional service account to impersonate using short-term
+        credentials, or chained list of accounts required to get the access_token
+        of the last account in the list, which will be impersonated in the request.
+        If set as a string, the account must grant the originating account
+        the Service Account Token Creator IAM role.
+        If set as a sequence, the identities from the list must grant
+        Service Account Token Creator IAM role to the directly preceding identity, with first
+        account from the list granting this role to the originating account (templated).
+    """
+
+    template_fields = ("location", "project_id", "impersonation_chain", "job_name", "gemini_api_key")
+
+    def __init__(
+        self,
+        *,
+        project_id: str,
+        location: str,
+        job_name: str,
+        gemini_api_key: str,
+        gcp_conn_id: str = "google_cloud_default",
+        impersonation_chain: str | Sequence[str] | None = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+
+        self.project_id = project_id
+        self.location = location
+        self.gcp_conn_id = gcp_conn_id
+        self.impersonation_chain = impersonation_chain
+        self.job_name = job_name
+        self.gemini_api_key = gemini_api_key
+
+    def execute(self, context: Context):
+        self.hook = GenAIGeminiAPIHook(
+            gcp_conn_id=self.gcp_conn_id,
+            impersonation_chain=self.impersonation_chain,
+            gemini_api_key=self.gemini_api_key,
+        )
+
+        try:
+            self.hook.delete_batch_job(job_name=self.job_name)
```

Comment:
As I mentioned before the request returns A DeleteJob with [deletion status](https://github.com/googleapis/python-genai/blob/main/google/genai/batches.py#L1841). It means for me that request creates a delete operation only and maybe here we need to wait until the Batch will be removed completely WDYT?

Reply:
Can you please mention this in the doc that it is not actually operator problem and API trick?